internal/generate/clusterserviceversion/clusterserviceversion_updaters.go

```diff
@@ -395,8 +407,138 @@ func mutatingToWebhookDescription(webhook admissionregv1.MutatingWebhook, depNam
 	return description
 }
 
+// The default AdmissionReviewVersions set in a CSV for conversion webhooks
+// if ConversionReviewVersions is not specified in CRDs.
+var defaultConversionReviewVersions = []string{"v1beta1", "v1"}
+
+// conversionToWebhookDescription builds a webhookDescription for conversion webhook with the
+// WebhookClientConfig specified in either v1 or v1beta1 CRDs.
+func conversionToWebhookDescription(crdNames []string, v1WebhookClientConfig *apiextv1.WebhookConversion, v1beta1WebhookClientConfig *apiextv1beta1.CustomResourceConversion, depName string, ws *corev1.Service) operatorsv1alpha1.WebhookDescription {
+
+	conversionReviewVersions, whPath, serviceRefName, serviceRefPort := getConversionWebhookDetails(v1WebhookClientConfig, v1beta1WebhookClientConfig)
+
+	description := operatorsv1alpha1.WebhookDescription{
+		Type:                    operatorsv1alpha1.ConversionWebhook,
+		ConversionCRDs:          crdNames,
+		AdmissionReviewVersions: conversionReviewVersions,
+		WebhookPath:             &whPath,
+		DeploymentName:          depName,
+		SideEffects: func() *admissionregv1.SideEffectClass {
+			seNone := admissionregv1.SideEffectClassNone
+			return &seNone
+		}(),
+	}
+
+	if len(description.AdmissionReviewVersions) == 0 {
+		description.AdmissionReviewVersions = defaultConversionReviewVersions
+	}
+
+	var webhookServiceRefPort int32 = 443
+
+	if serviceRefPort != nil {
+		webhookServiceRefPort = *serviceRefPort
+	}
+
+	if ws != nil {
+		for _, port := range ws.Spec.Ports {
+			if webhookServiceRefPort == port.Port {
+				description.ContainerPort = port.Port
+				description.TargetPort = &port.TargetPort
+				break
+			}
+		}
+	}
+
+	if description.DeploymentName == "" {
+		description.DeploymentName = strings.TrimSuffix(serviceRefName, "-service")
+	}
+	return description
+}
+
+// getConvWebhookCRDNamesAndConfig returns the list of the CRDs which have conversion webhookClient config
+// specified. It also returns the WebhookConversion config for v1 and v1beta1 CRDs. Getting the WebhookConversion Config
+// from the CRDs is necessary to extract the serviceRef specifications for webhook.
+func getConvWebhookCRDNamesAndConfig(c *collector.Manifests, serviceName string) ([]string, *apiextv1.WebhookConversion, *apiextv1beta1.CustomResourceConversion) {
+	if serviceName == "" {
+		return nil, nil, nil
+	}
+
+	var crdNames []string
+	var v1CRDWebhookConfig *apiextv1.WebhookConversion
+	var v1beta1WebhookConfig *apiextv1beta1.CustomResourceConversion
+
+	for _, crd := range c.V1CustomResourceDefinitions {
+		if crd.Spec.Conversion != nil {
+			whConv := crd.Spec.Conversion.Webhook
+			if whConv != nil && whConv.ClientConfig != nil && whConv.ClientConfig.Service != nil {
+				if whConv.ClientConfig.Service.Name == serviceName {
+					crdNames = append(crdNames, crd.GetName())
+					v1CRDWebhookConfig = whConv
```

Comment:
What if two CRDs have different conversion configs?

Reply:
The one in v1CRD is given priority. Both the configs would have the same serviceReference name. But if port and path differ, we may have to choose one.

Reply:
You can have one service serve multiple conversion webhooks for different CRDs, regardless of CRD version. Right now this code chooses the last conversion config found, resulting in a potentially incorrect config for CRDs with a corresponding service name earlier in the loop if the scenario I suggest is the current case.